From 0c761f7bf27c81d67b01bc7e6ace9f7e04baf6c4 Mon Sep 17 00:00:00 2001
From: Daniel Stone <daniels@collabora.com>
Date: Mon, 24 Oct 2016 13:26:42 +0100
Subject: [PATCH 52/94] compositor-drm: Rename fb_plane to scanout_plane

All planes being displayed have a framebuffer. What makes 'fb_plane'
special is that it's being displayed as the primary plane by KMS.

Previous patchsets renamed this to 'primary_plane' to match the KMS
terminology, namely the CRTC's base plane, which is controlled by
drmModeSetCrtc in the legacy API, and identified by PLANE_TYPE ==
"Primary" in the universal-plane API.

However, Weston uses 'primary_plane' internally to refer to the case
where client content is _not_ directly displayed on a plane, but
composited via the renderer, with the result of the compositing then
shown.

Rename to 'scanout_plane' as our least-ambiguous name, and document it a
bit.

Signed-off-by: Daniel Stone <daniels@collabora.com>

Differential Revision: https://phabricator.freedesktop.org/D1415
---
 src/compositor-drm.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index 4222753..4f08ae1 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -230,7 +230,8 @@ struct drm_output {
 	struct gbm_surface *gbm_surface;
 	uint32_t gbm_format;
 
-	struct weston_plane fb_plane;
+	/* Plane currently being directly displayed by KMS */
+	struct weston_plane scanout_plane;
 	struct drm_fb *fb_current, *fb_pending, *fb_last;
 
 	struct drm_fb *dumb[2];
@@ -635,7 +636,7 @@ drm_output_prepare_scanout_view(struct drm_output *output,
 
 	drm_fb_set_buffer(output->fb_pending, buffer);
 
-	return &output->fb_plane;
+	return &output->scanout_plane;
 }
 
 static struct drm_fb *
@@ -1498,7 +1499,7 @@ drm_output_destroy(struct weston_output *output_base)
 	else
 		drm_output_fini_egl(output);
 
-	weston_plane_release(&output->fb_plane);
+	weston_plane_release(&output->scanout_plane);
 	weston_plane_release(&output->cursor_plane);
 
 	weston_output_destroy(&output->base);
@@ -2731,10 +2732,10 @@ create_output_for_connector(struct drm_backend *b,
 
 	weston_plane_init(&output->cursor_plane, b->compositor,
 			  INT32_MIN, INT32_MIN);
-	weston_plane_init(&output->fb_plane, b->compositor, 0, 0);
+	weston_plane_init(&output->scanout_plane, b->compositor, 0, 0);
 
 	weston_compositor_stack_plane(b->compositor, &output->cursor_plane, NULL);
-	weston_compositor_stack_plane(b->compositor, &output->fb_plane,
+	weston_compositor_stack_plane(b->compositor, &output->scanout_plane,
 				      &b->compositor->primary_plane);
 
 	weston_log("Output %s, (connector %d, crtc %d)\n",
-- 
1.9.1

